pkg/agent/openflow/network_policy.go

```diff
@@ -661,56 +668,61 @@ func generateAddressConjMatch(ruleTableID uint8, addr types.Address, addrType ty
 	return match
 }
 
-func getServiceMatchType(protocol *v1beta2.Protocol, ipv4Enabled, ipv6Enabled, matchSrc bool) []*types.MatchKey {
+func getServiceMatchType(protocol *v1beta2.Protocol, ipProtocols []binding.Protocol, matchSrc bool) []*types.MatchKey {
 	var matchKeys []*types.MatchKey
 	switch *protocol {
 	case v1beta2.ProtocolTCP:
 		if !matchSrc {
-			if ipv4Enabled {
-				matchKeys = append(matchKeys, MatchTCPDstPort)
-			}
-			if ipv6Enabled {
-				matchKeys = append(matchKeys, MatchTCPv6DstPort)
+			for _, ipProtocol := range ipProtocols {
+				if ipProtocol == binding.ProtocolIP {
+					matchKeys = append(matchKeys, MatchTCPDstPort)
+				} else if ipProtocol == binding.ProtocolIPv6 {
+					matchKeys = append(matchKeys, MatchTCPv6DstPort)
+				}
 			}
 		} else {
-			if ipv4Enabled {
-				matchKeys = append(matchKeys, MatchTCPSrcPort)
-			}
-			if ipv6Enabled {
-				matchKeys = append(matchKeys, MatchTCPv6SrcPort)
+			for _, ipProtocol := range ipProtocols {
+				if ipProtocol == binding.ProtocolIP {
+					matchKeys = append(matchKeys, MatchTCPSrcPort)
+				} else if ipProtocol == binding.ProtocolIPv6 {
+					matchKeys = append(matchKeys, MatchTCPv6SrcPort)
+				}
 			}
 		}
 	case v1beta2.ProtocolUDP:
 		if !matchSrc {
-			if ipv4Enabled {
-				matchKeys = append(matchKeys, MatchUDPDstPort)
-			}
-			if ipv6Enabled {
-				matchKeys = append(matchKeys, MatchUDPv6DstPort)
+			for _, ipProtocol := range ipProtocols {
+				if ipProtocol == binding.ProtocolIP {
+					matchKeys = append(matchKeys, MatchUDPDstPort)
+				} else if ipProtocol == binding.ProtocolIPv6 {
+					matchKeys = append(matchKeys, MatchUDPv6DstPort)
+				}
 			}
```

Comment:
Yeah, but previous code is like this. I don't want to refactor more detailed code of NP in this PR. I think this can be done in a further PR which refactors NP code.